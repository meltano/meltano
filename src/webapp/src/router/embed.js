import Router from 'vue-router'

import ResourceEmbed from '@/views/ResourceEmbed'

const router = new Router({
  mode: 'history',
  routes: [
    {
      path: '/-/embed/:token',
      name: 'resource-embed',
      component: ResourceEmbed,
      meta: {
        title: 'Meltano Resource Embed',
      },
      props: (route) => ({ ...route.params, today: route.query.today }),
    },
  ],
})

export default router
