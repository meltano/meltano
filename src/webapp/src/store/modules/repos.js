import lodash from 'lodash'

import utils from '@/utils/utils'
import repoApi from '../../api/repo'

const defaultState = utils.deepFreeze({
  activeView: {
    is_markdown: false,
    file: '',
    populated: false
  },
  loadingValidation: false,
  loadingUpdate: false,
  models: {},
  validated: false,
  errors: [],
  files: {}
})

const getters = {
  hasMarkdown(state) {
    return state.activeView.populated && state.activeView.is_markdown
  },

  urlForModelDesign: () => (model, design) => `/analyze/${model}/${design}`,

  hasCode(state) {
    return state.activeView.populated && !state.activeView.is_markdown
  },

  hasError(state) {
    return state.errors && state.errors.length
  },

  hasFiles(state) {
    return state.files.hasOwnProperty('topics') && state.files.topics.items
  },

  hasModels(state) {
    return !lodash.isEmpty(state.models)
  },

  passedValidation(state) {
    return state.validated && state.errors && !state.errors.length
  }
}

const actions = {
  getRepo({ state, commit }) {
    repoApi.index().then(response => {
      const files = response.data
      commit('setValidatedState', response.data)
      state.loadingValidation = false
      commit('setRepoFiles', { files })
    })
  },

  getFile({ commit }, file) {
    repoApi.file(file.id).then(response => {
      commit('setCurrentFileTable', response.data)
    })
  },

  lint({ state, commit }) {
    state.loadingValidation = true
    repoApi
      .lint()
      .then(response => {
        commit('setValidatedState', response.data)
        state.loadingValidation = false
      })
      .catch(() => {
        state.loadingValidation = false
      })
  },

  sync({ state, commit, dispatch }) {
    state.loadingUpdate = true
    repoApi
      .sync()
      .then(response => {
        dispatch('getModels')
        commit('setValidatedState', response.data)
        state.loadingUpdate = false
      })
      .catch(() => {
        state.loadingUpdate = false
      })
  },

  getModels({ commit }) {
    repoApi.models().then(response => {
      commit('setModels', response.data)
    })
  }
}

const mutations = {
  setModels(state, models) {
    state.models = models
  },

  setRepoFiles(state, { files }) {
    state.files = files
  },

  setCurrentFileTable(state, file) {
    state.activeView = file
  },

  setValidatedState(state, validated) {
    state.errors = []
    state.validated = true
    // validation failed, so there will be errors
    if (!validated.result) {
      state.errors = validated.errors
    }
  }
}

export default {
  namespaced: true,
  state: lodash.cloneDeep(defaultState),
  getters,
  actions,
  mutations
}
