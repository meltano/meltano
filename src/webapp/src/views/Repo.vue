<script>
import { mapState, mapGetters } from 'vuex'
import fileTypeEnums from '@/utils/fileTypeEnums'
import pretty from '@/filters/pretty'
import RouterViewLayout from '@/views/RouterViewLayout'
import utils from '@/utils/utils'

export default {
  name: 'Repo',
  created() {
    this.getRepo()
    this.sync()
  },
  components: {
    RouterViewLayout
  },
  filters: {
    pretty
  },
  computed: {
    ...mapGetters('repos', [
      'hasFiles',
      'hasError',
      'passedValidation',
      'hasMarkdown',
      'hasCode'
    ]),
    ...mapState('repos', [
      'files',
      'activeView',
      'validated',
      'loadingValidation',
      'loadingUpdate',
      'errors'
    ])
  },
  methods: {
    getRepo() {
      this.$store.dispatch('repos/getRepo')
    },
    jsDashify(type, name) {
      return utils.jsDashify(type, name)
    },
    isActive(f) {
      return f.id === this.activeView.id
    },
    isDeepRoutable(type) {
      return type === fileTypeEnums.dashboards || type === fileTypeEnums.reports
    },
    getDeepRoute(key, file) {
      const name = utils.capitalize(utils.singularize(key))
      const params = { slug: file.slug }
      if (file.model && file.design) {
        params.model = file.model
        params.design = file.design
      }
      return { name, params }
    },
    getFile(file) {
      this.$store.dispatch('repos/getFile', file)
    },
    lint() {
      this.$store.dispatch('repos/lint')
    },
    sync() {
      this.$store.dispatch('repos/sync')
    }
  }
}
</script>

<template>
  <router-view-layout>
    <div class="container view-header">
      <div class="content">
        <div class="level">
          <h1 class="is-marginless">Repo</h1>
        </div>
      </div>
    </div>

    <div class="container view-body">
      <section>
        <div class="columns is-gapless">
          <aside class="column is-one-quarter">
            <div class="level">
              <div class="level-left">
                <div class="field has-addons">
                  <div class="control">
                    <a
                      href="#"
                      class="button is-small"
                      :class="{ 'is-loading': loadingValidation }"
                      @click="lint"
                      >Lint</a
                    >
                  </div>
                  <div class="control">
                    <a
                      href="#"
                      class="button is-small"
                      :class="{ 'is-loading': loadingUpdate }"
                      @click="sync"
                      >Sync</a
                    >
                  </div>
                </div>
              </div>
              <div class="level-right">
                <span class="tag is-success pull-right" v-if="passedValidation"
                  >Passed!</span
                >
                <span class="tag is-warning" v-if="!validated"
                  >Unvalidated</span
                >
                <span class="tag is-danger" v-if="hasError">Errors</span>
              </div>
            </div>
            <template v-if="hasError">
              <nav class="panel">
                <!-- eslint-disable-next-line vue/require-v-for-key -->
                <div
                  class="panel-block has-background-white"
                  v-for="err in errors"
                >
                  <ul>
                    <li class="level">
                      <div class="tags has-addons">
                        <span class="tag is-info">?</span>
                        <span class="tag">{{ err.fileName }}</span>
                      </div>
                    </li>
                    <li class="error-desc-cont">
                      <code class="error-desc">{{ err.message }}</code>
                    </li>
                  </ul>
                </div>
              </nav>
            </template>
            <template v-if="!hasFiles">
              <p class="menu-label">
                No files found
              </p>
            </template>
            <template v-for="(value, key) in files">
              <!-- eslint-disable-next-line vue/require-v-for-key -->
              <p class="menu-label">
                {{ value.label }}
              </p>
              <!-- eslint-disable-next-line vue/require-v-for-key -->
              <ul class="menu-list">
                <template v-if="value.items">
                  <li v-for="file in value.items" :key="file.abs">
                    <div class="columns is-vcentered">
                      <div class="column">
                        <a
                          :class="[
                            { 'is-active': isActive(file) },
                            jsDashify(value.label, file.name)
                          ]"
                          @click.prevent="getFile(file)"
                        >
                          {{ file.name }}
                        </a>
                      </div>
                      <div
                        v-if="isDeepRoutable(key)"
                        class="column is-one-fifth"
                      >
                        <router-link
                          :to="getDeepRoute(key, file)"
                          class="button is-secondary is-light is-small is-pulled-right"
                        >
                          <!-- TODO temporary icon, find better solution -->
                          <font-awesome-icon icon="arrow-right" />
                        </router-link>
                      </div>
                    </div>
                  </li>
                </template>
              </ul>
            </template>
          </aside>
          <div class="column is-three-quarters">
            <div v-if="!activeView.populated">
              <div
                class="empty-state
                has-background-white
                has-text-centered
                is-size-4
                is-uppercase"
              >
                Select a file
              </div>
            </div>
            <div v-if="hasMarkdown">
              <div
                class="js-markdown-preview has-background-white"
                v-html="activeView.file"
              ></div>
            </div>
            <div
              class="js-code-preview is-paddingless code-container"
              v-else-if="hasCode"
            >
              <div class="has-background-white">
                <pre>{{ activeView.file | pretty }}</pre>
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
  </router-view-layout>
</template>

<style lang="scss" scoped>
.content {
  padding: 10px;
  position: fixed;
  top: 52px;
  left: 430px;
  right: 0;
  bottom: 0;
  overflow: scroll;
  z-index: 1;
}

.code-container {
  // to overcome the important on paddingless
  padding-left: 1rem !important;
}

.error-desc-cont {
  .tag {
    margin-bottom: 0.5rem;
  }
  .error-desc {
    display: block;
  }
}

.empty-state {
  padding-top: 200px;
}
</style>
