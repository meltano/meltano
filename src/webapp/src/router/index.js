import Router from 'vue-router'

import AnalyzeModels from '@/components/analyze/AnalyzeModels'
import AnalyzeSettings from '@/components/analyze/AnalyzeSettings'
import CreateScheduleModal from '@/components/pipelines/CreateScheduleModal'
import Design from '@/components/analyze/Design'
import EntitiesSelectorModal from '@/components/pipelines/EntitiesSelectorModal'
import Entities from '@/components/pipelines/Entities'
import Extractors from '@/components/pipelines/Extractors'
import ExtractorSettingsModal from '@/components/pipelines/ExtractorSettingsModal'
import Loaders from '@/components/pipelines/Loaders'
import LoaderSettingsModal from '@/components/pipelines/LoaderSettingsModal'
import PipelineSchedules from '@/components/pipelines/PipelineSchedules'

import Dashboards from '@/views/Dashboards'
import Analyze from '@/views/Analyze'
import NotFound from '@/views/NotFound'
import Orchestration from '@/views/Orchestration'
import Pipelines from '@/views/Pipelines'
import Repo from '@/views/Repo'

const router = new Router({
  mode: 'history',
  routes: [
    {
      path: '*',
      name: '404',
      component: NotFound
    },
    {
      path: '/',
      redirect: '/pipelines'
    },
    {
      path: '/pipelines/',
      redirect: '/pipelines/extractors/',
      name: 'dataSetup',
      component: Pipelines,
      children: [
        {
          path: 'extractors',
          name: 'extractors',
          components: {
            default: Extractors
          },
          meta: {
            isModal: false
          }
        },
        {
          path: 'extractors/:extractor',
          name: 'extractorSettings',
          components: {
            default: Extractors,
            extractorSettings: ExtractorSettingsModal
          },
          meta: {
            isModal: true
          }
        },
        {
          path: 'entities',
          name: 'entities',
          components: {
            default: Entities
          },
          meta: {
            isModal: false
          }
        },
        {
          path: 'entities/:extractor',
          name: 'extractorEntities',
          components: {
            default: Entities,
            extractorEntities: EntitiesSelectorModal
          },
          meta: {
            isModal: true
          }
        },
        {
          path: 'loaders',
          name: 'loaders',
          components: {
            default: Loaders
          },
          meta: {
            isModal: false
          }
        },
        {
          path: 'loaders/:loader',
          name: 'loaderSettings',
          components: {
            default: Loaders,
            loaderSettings: LoaderSettingsModal
          },
          meta: {
            isModal: true
          }
        },
        {
          path: 'schedules',
          name: 'schedules',
          components: {
            default: PipelineSchedules
          },
          meta: {
            isModal: false
          }
        },
        {
          path: 'schedules/create',
          name: 'createSchedule',
          components: {
            default: PipelineSchedules,
            createSchedule: CreateScheduleModal
          },
          meta: {
            isModal: true
          }
        }
      ]
    },
    {
      path: '/orchestration/',
      name: 'orchestration',
      component: Orchestration
    },
    {
      path: '/files/',
      name: 'projectFiles',
      component: Repo
    },
    {
      path: '/analyze/',
      redirect: '/analyze/models/',
      name: 'analyze',
      component: Analyze,
      children: [
        {
          path: 'models',
          name: 'analyzeModels',
          component: AnalyzeModels
        },
        {
          path: 'settings',
          name: 'analyzeSettings',
          component: AnalyzeSettings
        },
        {
          path: '/analyze/:model/:design',
          name: 'analyzeDesign',
          component: Design
        },
        {
          path: '/analyze/:model/:design/reports/report/:slug',
          name: 'Report',
          component: Design
        }
      ]
    },
    {
      path: '/dashboards/',
      name: 'Dashboards',
      component: Dashboards
    },
    {
      path: '/dashboards/dashboard/:slug',
      name: 'Dashboard',
      component: Dashboards
    }
  ]
})

export default router
