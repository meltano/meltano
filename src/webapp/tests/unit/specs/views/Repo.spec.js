import Vuex from 'vuex'
import Router from 'vue-router'
import { mount, createLocalVue, shallowMount } from '@vue/test-utils'
import Repo from '@/views/Repo'
import repos from '@/store/modules/repos'
import router from '@/router'

const localVue = createLocalVue()
localVue.use(Vuex)
localVue.use(Router)

describe('Repo.vue', () => {
  let actions
  let state
  let store

  const createShallowWrapper = () =>
    shallowMount(Repo, {
      store,
      localVue,
      router
    })

  const createWrapper = () =>
    mount(Repo, {
      store,
      localVue,
      router,
      stubs: ['font-awesome-icon']
    })

  beforeEach(() => {
    state = repos.state
    actions = {
      getRepo: jest.fn(),
      getFile: jest.fn(),
      sync: jest.fn()
    }
    store = new Vuex.Store({
      modules: {
        repos: {
          namespaced: true,
          state,
          actions,
          getters: repos.getters
        }
      }
    })
  })

  it('calls getRepo() and sync() via created() lifecycle hook', () => {
    const wrapper = createShallowWrapper()

    expect(wrapper.html()).toBeTruthy()
    expect(actions.getRepo).toHaveBeenCalled()
    expect(actions.sync).toHaveBeenCalled()
  })

  it('renders no code or markdown by default', () => {
    const wrapper = createWrapper()

    expect(wrapper.element).toMatchSnapshot()
  })

  it('renders markdown in the preview pane for markdown files', () => {
    state.activeView = {
      is_markdown: true,
      file: '<h1>Title</h1>',
      populated: true
    }
    const wrapper = createWrapper()

    expect(wrapper.element).toMatchSnapshot()
  })

  it('renders code in the preview pane for code files', () => {
    state.activeView = {
      is_markdown: false,
      file: '{ "title": "Title" }',
      populated: true
    }
    const wrapper = createWrapper()

    expect(wrapper.element).toMatchSnapshot()
  })

  it('renders the dashboard list when dashboards exist and calls getFile() when a dashboard is clicked', () => {
    state.files = {
      dashboards: {
        label: 'Dashboards',
        items: [
          {
            createdAt: 1551459531.431577,
            description: '',
            id:
              'F5KXGZLSOMXWI23ON54C2Z3JORWGCYRPIRXWG5LNMVXHI4ZPKBZG62TFMN2HGL3DMFZGE33OFVTWS5DGNRUXQL3NN5SGK3BPMIXGIYLTNBRG6YLSMQXG2NLP',
            name: 'Some',
            path: 'some.dashboard.m5o',
            reportIds: [],
            slug: 'some',
            version: '1.0.0'
          }
        ]
      }
    }
    const wrapper = createWrapper()

    expect(wrapper.element).toMatchSnapshot()
    wrapper.find('.js-dashboards-some').trigger('click')
    expect(actions.getFile).toHaveBeenCalled()
  })

  it('renders the document list when documents exist and calls getFile() when a document is clicked', () => {
    state.files = {
      documents: {
        label: 'Documents',
        items: [
          {
            createdAt: 1551990614.693281,
            id:
              'F5KXGZLSOMXWI23ON54C2Z3JORWGCYRPIRXWG5LNMVXHI4ZPKBZG62TFMN2HGL3DMFZGE33OFVTWS5DGNRUXQL2SIVAUITKFFZWWI===',
            name: 'README',
            path: 'README.md',
            slug: 'readme-md'
          }
        ]
      }
    }
    const wrapper = createWrapper()

    expect(wrapper.element).toMatchSnapshot()
    wrapper.find('.js-documents-readme').trigger('click')
    expect(actions.getFile).toHaveBeenCalled()
  })

  it('renders the report list when reports exist and calls getFile() when a report is clicked', () => {
    state.files = {
      reports: {
        label: 'Reports',
        items: [
          {
            chartType: 'BarChart',
            createdAt: 1551461064.508327,
            design: 'region',
            id:
              'F5KXGZLSOMXWI23ON54C2Z3JORWGCYRPIRXWG5LNMVXHI4ZPKBZG62TFMN2HGL3DMFZGE33OFVTWS5DGNRUXQL3NN5SGK3BPMMZC44TFOBXXE5BONU2W6===',
            model: 'carbon',
            name: 'Some',
            path: 'some.report.m5o',
            queryPayload: {},
            slug: 'some',
            version: '1.0.0'
          }
        ]
      }
    }
    const wrapper = createWrapper()

    expect(wrapper.element).toMatchSnapshot()
    wrapper.find('.js-reports-some').trigger('click')
    expect(actions.getFile).toHaveBeenCalled()
  })

  it('renders the table list when tables exist and calls getFile() when a table is clicked', () => {
    state.files = {
      tables: {
        label: 'Tables',
        items: [
          {
            createdAt: 1551989728.153954,
            id: 'MVYGS43PMRSXGLTUMFRGYZJONU2W6===',
            name: 'episodes',
            path: 'episodes.table.m5o',
            slug: 'episodes'
          }
        ]
      }
    }
    const wrapper = createWrapper()

    expect(wrapper.element).toMatchSnapshot()
    wrapper.find('.js-tables-episodes').trigger('click')
    expect(actions.getFile).toHaveBeenCalled()
  })

  it('renders the topic list when topics exist and calls getFile() when a topic is clicked', () => {
    state.files = {
      topics: {
        label: 'Topics',
        items: [
          {
            version: 1,
            name: 'carbon',
            connection: 'runners_db',
            label: 'carbon intensity',
            designs: {}
          }
        ]
      }
    }
    const wrapper = createWrapper()

    expect(wrapper.element).toMatchSnapshot()
    wrapper.find('.js-topics-carbon').trigger('click')
    expect(actions.getFile).toHaveBeenCalled()
  })
})
