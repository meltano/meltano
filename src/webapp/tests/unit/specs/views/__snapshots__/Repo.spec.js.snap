// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repo.vue renders code in the preview pane for code files 1`] = `
<div
  class="columns is-gapless"
>
  <section
    class="column is-full"
  >
    <div
      class="box-transparent box is-radiusless is-shadowless"
    >
      <div
        class="container view-header"
      >
        <div
          class="content"
        >
          <div
            class="level"
          >
            <h1
              class="is-marginless"
            >
              Repo
            </h1>
          </div>
        </div>
      </div>
       
      <div
        class="container view-body"
      >
        <section>
          <div
            class="columns is-gapless"
          >
            <aside
              class="column is-one-quarter"
            >
              <div
                class="level"
              >
                <div
                  class="level-left"
                >
                  <div
                    class="field has-addons"
                  >
                    <div
                      class="control"
                    >
                      <a
                        class="button is-small"
                        href="#"
                      >
                        Lint
                      </a>
                    </div>
                     
                    <div
                      class="control"
                    >
                      <a
                        class="button is-small"
                        href="#"
                      >
                        Sync
                      </a>
                    </div>
                  </div>
                </div>
                 
                <div
                  class="level-right"
                >
                  <!---->
                   
                  <span
                    class="tag is-warning"
                  >
                    Unvalidated
                  </span>
                   
                  <!---->
                </div>
              </div>
               
              <!---->
               
              <p
                class="menu-label"
              >
                
              No files found
            
              </p>
               
            </aside>
             
            <div
              class="column is-three-quarters"
            >
              <!---->
               
              <div
                class="js-code-preview is-paddingless code-container"
              >
                <div
                  class="has-background-white"
                >
                  <pre>
                    {
  "title": "Title"
}
                  </pre>
                </div>
              </div>
            </div>
          </div>
        </section>
      </div>
    </div>
  </section>
</div>
`;

exports[`Repo.vue renders markdown in the preview pane for markdown files 1`] = `
<div
  class="columns is-gapless"
>
  <section
    class="column is-full"
  >
    <div
      class="box-transparent box is-radiusless is-shadowless"
    >
      <div
        class="container view-header"
      >
        <div
          class="content"
        >
          <div
            class="level"
          >
            <h1
              class="is-marginless"
            >
              Repo
            </h1>
          </div>
        </div>
      </div>
       
      <div
        class="container view-body"
      >
        <section>
          <div
            class="columns is-gapless"
          >
            <aside
              class="column is-one-quarter"
            >
              <div
                class="level"
              >
                <div
                  class="level-left"
                >
                  <div
                    class="field has-addons"
                  >
                    <div
                      class="control"
                    >
                      <a
                        class="button is-small"
                        href="#"
                      >
                        Lint
                      </a>
                    </div>
                     
                    <div
                      class="control"
                    >
                      <a
                        class="button is-small"
                        href="#"
                      >
                        Sync
                      </a>
                    </div>
                  </div>
                </div>
                 
                <div
                  class="level-right"
                >
                  <!---->
                   
                  <span
                    class="tag is-warning"
                  >
                    Unvalidated
                  </span>
                   
                  <!---->
                </div>
              </div>
               
              <!---->
               
              <p
                class="menu-label"
              >
                
              No files found
            
              </p>
               
            </aside>
             
            <div
              class="column is-three-quarters"
            >
              <!---->
               
              <div>
                <div
                  class="js-markdown-preview has-background-white"
                >
                  <h1>
                    Title
                  </h1>
                </div>
              </div>
            </div>
          </div>
        </section>
      </div>
    </div>
  </section>
</div>
`;

exports[`Repo.vue renders no code or markdown by default 1`] = `
<div
  class="columns is-gapless"
>
  <section
    class="column is-full"
  >
    <div
      class="box-transparent box is-radiusless is-shadowless"
    >
      <div
        class="container view-header"
      >
        <div
          class="content"
        >
          <div
            class="level"
          >
            <h1
              class="is-marginless"
            >
              Repo
            </h1>
          </div>
        </div>
      </div>
       
      <div
        class="container view-body"
      >
        <section>
          <div
            class="columns is-gapless"
          >
            <aside
              class="column is-one-quarter"
            >
              <div
                class="level"
              >
                <div
                  class="level-left"
                >
                  <div
                    class="field has-addons"
                  >
                    <div
                      class="control"
                    >
                      <a
                        class="button is-small"
                        href="#"
                      >
                        Lint
                      </a>
                    </div>
                     
                    <div
                      class="control"
                    >
                      <a
                        class="button is-small"
                        href="#"
                      >
                        Sync
                      </a>
                    </div>
                  </div>
                </div>
                 
                <div
                  class="level-right"
                >
                  <!---->
                   
                  <span
                    class="tag is-warning"
                  >
                    Unvalidated
                  </span>
                   
                  <!---->
                </div>
              </div>
               
              <!---->
               
              <p
                class="menu-label"
              >
                
              No files found
            
              </p>
               
            </aside>
             
            <div
              class="column is-three-quarters"
            >
              <div>
                <div
                  class="empty-state
              has-background-white
              has-text-centered
              is-size-4
              is-uppercase"
                >
                  
              Select a file
            
                </div>
              </div>
               
              <!---->
            </div>
          </div>
        </section>
      </div>
    </div>
  </section>
</div>
`;

exports[`Repo.vue renders the dashboard list when dashboards exist and calls getFile() when a dashboard is clicked 1`] = `
<div
  class="columns is-gapless"
>
  <section
    class="column is-full"
  >
    <div
      class="box-transparent box is-radiusless is-shadowless"
    >
      <div
        class="container view-header"
      >
        <div
          class="content"
        >
          <div
            class="level"
          >
            <h1
              class="is-marginless"
            >
              Repo
            </h1>
          </div>
        </div>
      </div>
       
      <div
        class="container view-body"
      >
        <section>
          <div
            class="columns is-gapless"
          >
            <aside
              class="column is-one-quarter"
            >
              <div
                class="level"
              >
                <div
                  class="level-left"
                >
                  <div
                    class="field has-addons"
                  >
                    <div
                      class="control"
                    >
                      <a
                        class="button is-small"
                        href="#"
                      >
                        Lint
                      </a>
                    </div>
                     
                    <div
                      class="control"
                    >
                      <a
                        class="button is-small"
                        href="#"
                      >
                        Sync
                      </a>
                    </div>
                  </div>
                </div>
                 
                <div
                  class="level-right"
                >
                  <!---->
                   
                  <span
                    class="tag is-warning"
                  >
                    Unvalidated
                  </span>
                   
                  <!---->
                </div>
              </div>
               
              <!---->
               
              <p
                class="menu-label"
              >
                
              No files found
            
              </p>
               
              <p
                class="menu-label"
              >
                
              Dashboards
            
              </p>
               
              <ul
                class="menu-list"
              >
                <li>
                  <div
                    class="columns is-vcentered"
                  >
                    <div
                      class="column"
                    >
                      <a
                        class="js-dashboards-some"
                      >
                        
                        Some
                      
                      </a>
                    </div>
                     
                    <div
                      class="column is-one-fifth"
                    >
                      <a
                        class="button is-secondary is-light is-small is-pulled-right"
                        href="/dashboards/dashboard/some"
                      >
                        <font-awesome-icon-stub
                          icon="arrow-right"
                        />
                      </a>
                    </div>
                  </div>
                </li>
              </ul>
            </aside>
             
            <div
              class="column is-three-quarters"
            >
              <!---->
               
              <div
                class="js-code-preview is-paddingless code-container"
              >
                <div
                  class="has-background-white"
                >
                  <pre>
                    {
  "title": "Title"
}
                  </pre>
                </div>
              </div>
            </div>
          </div>
        </section>
      </div>
    </div>
  </section>
</div>
`;

exports[`Repo.vue renders the document list when documents exist and calls getFile() when a document is clicked 1`] = `
<div
  class="columns is-gapless"
>
  <section
    class="column is-full"
  >
    <div
      class="box-transparent box is-radiusless is-shadowless"
    >
      <div
        class="container view-header"
      >
        <div
          class="content"
        >
          <div
            class="level"
          >
            <h1
              class="is-marginless"
            >
              Repo
            </h1>
          </div>
        </div>
      </div>
       
      <div
        class="container view-body"
      >
        <section>
          <div
            class="columns is-gapless"
          >
            <aside
              class="column is-one-quarter"
            >
              <div
                class="level"
              >
                <div
                  class="level-left"
                >
                  <div
                    class="field has-addons"
                  >
                    <div
                      class="control"
                    >
                      <a
                        class="button is-small"
                        href="#"
                      >
                        Lint
                      </a>
                    </div>
                     
                    <div
                      class="control"
                    >
                      <a
                        class="button is-small"
                        href="#"
                      >
                        Sync
                      </a>
                    </div>
                  </div>
                </div>
                 
                <div
                  class="level-right"
                >
                  <!---->
                   
                  <span
                    class="tag is-warning"
                  >
                    Unvalidated
                  </span>
                   
                  <!---->
                </div>
              </div>
               
              <!---->
               
              <p
                class="menu-label"
              >
                
              No files found
            
              </p>
               
              <p
                class="menu-label"
              >
                
              Documents
            
              </p>
               
              <ul
                class="menu-list"
              >
                <li>
                  <div
                    class="columns is-vcentered"
                  >
                    <div
                      class="column"
                    >
                      <a
                        class="js-documents-readme"
                      >
                        
                        README
                      
                      </a>
                    </div>
                     
                    <!---->
                  </div>
                </li>
              </ul>
            </aside>
             
            <div
              class="column is-three-quarters"
            >
              <!---->
               
              <div
                class="js-code-preview is-paddingless code-container"
              >
                <div
                  class="has-background-white"
                >
                  <pre>
                    {
  "title": "Title"
}
                  </pre>
                </div>
              </div>
            </div>
          </div>
        </section>
      </div>
    </div>
  </section>
</div>
`;

exports[`Repo.vue renders the report list when reports exist and calls getFile() when a report is clicked 1`] = `
<div
  class="columns is-gapless"
>
  <section
    class="column is-full"
  >
    <div
      class="box-transparent box is-radiusless is-shadowless"
    >
      <div
        class="container view-header"
      >
        <div
          class="content"
        >
          <div
            class="level"
          >
            <h1
              class="is-marginless"
            >
              Repo
            </h1>
          </div>
        </div>
      </div>
       
      <div
        class="container view-body"
      >
        <section>
          <div
            class="columns is-gapless"
          >
            <aside
              class="column is-one-quarter"
            >
              <div
                class="level"
              >
                <div
                  class="level-left"
                >
                  <div
                    class="field has-addons"
                  >
                    <div
                      class="control"
                    >
                      <a
                        class="button is-small"
                        href="#"
                      >
                        Lint
                      </a>
                    </div>
                     
                    <div
                      class="control"
                    >
                      <a
                        class="button is-small"
                        href="#"
                      >
                        Sync
                      </a>
                    </div>
                  </div>
                </div>
                 
                <div
                  class="level-right"
                >
                  <!---->
                   
                  <span
                    class="tag is-warning"
                  >
                    Unvalidated
                  </span>
                   
                  <!---->
                </div>
              </div>
               
              <!---->
               
              <p
                class="menu-label"
              >
                
              No files found
            
              </p>
               
              <p
                class="menu-label"
              >
                
              Reports
            
              </p>
               
              <ul
                class="menu-list"
              >
                <li>
                  <div
                    class="columns is-vcentered"
                  >
                    <div
                      class="column"
                    >
                      <a
                        class="js-reports-some"
                      >
                        
                        Some
                      
                      </a>
                    </div>
                     
                    <div
                      class="column is-one-fifth"
                    >
                      <a
                        class="button is-secondary is-light is-small is-pulled-right"
                        href="/analyze/carbon/region/reports/report/some"
                      >
                        <font-awesome-icon-stub
                          icon="arrow-right"
                        />
                      </a>
                    </div>
                  </div>
                </li>
              </ul>
            </aside>
             
            <div
              class="column is-three-quarters"
            >
              <!---->
               
              <div
                class="js-code-preview is-paddingless code-container"
              >
                <div
                  class="has-background-white"
                >
                  <pre>
                    {
  "title": "Title"
}
                  </pre>
                </div>
              </div>
            </div>
          </div>
        </section>
      </div>
    </div>
  </section>
</div>
`;

exports[`Repo.vue renders the table list when tables exist and calls getFile() when a table is clicked 1`] = `
<div
  class="columns is-gapless"
>
  <section
    class="column is-full"
  >
    <div
      class="box-transparent box is-radiusless is-shadowless"
    >
      <div
        class="container view-header"
      >
        <div
          class="content"
        >
          <div
            class="level"
          >
            <h1
              class="is-marginless"
            >
              Repo
            </h1>
          </div>
        </div>
      </div>
       
      <div
        class="container view-body"
      >
        <section>
          <div
            class="columns is-gapless"
          >
            <aside
              class="column is-one-quarter"
            >
              <div
                class="level"
              >
                <div
                  class="level-left"
                >
                  <div
                    class="field has-addons"
                  >
                    <div
                      class="control"
                    >
                      <a
                        class="button is-small"
                        href="#"
                      >
                        Lint
                      </a>
                    </div>
                     
                    <div
                      class="control"
                    >
                      <a
                        class="button is-small"
                        href="#"
                      >
                        Sync
                      </a>
                    </div>
                  </div>
                </div>
                 
                <div
                  class="level-right"
                >
                  <!---->
                   
                  <span
                    class="tag is-warning"
                  >
                    Unvalidated
                  </span>
                   
                  <!---->
                </div>
              </div>
               
              <!---->
               
              <p
                class="menu-label"
              >
                
              No files found
            
              </p>
               
              <p
                class="menu-label"
              >
                
              Tables
            
              </p>
               
              <ul
                class="menu-list"
              >
                <li>
                  <div
                    class="columns is-vcentered"
                  >
                    <div
                      class="column"
                    >
                      <a
                        class="js-tables-episodes"
                      >
                        
                        episodes
                      
                      </a>
                    </div>
                     
                    <!---->
                  </div>
                </li>
              </ul>
            </aside>
             
            <div
              class="column is-three-quarters"
            >
              <!---->
               
              <div
                class="js-code-preview is-paddingless code-container"
              >
                <div
                  class="has-background-white"
                >
                  <pre>
                    {
  "title": "Title"
}
                  </pre>
                </div>
              </div>
            </div>
          </div>
        </section>
      </div>
    </div>
  </section>
</div>
`;

exports[`Repo.vue renders the topic list when topics exist and calls getFile() when a topic is clicked 1`] = `
<div
  class="columns is-gapless"
>
  <section
    class="column is-full"
  >
    <div
      class="box-transparent box is-radiusless is-shadowless"
    >
      <div
        class="container view-header"
      >
        <div
          class="content"
        >
          <div
            class="level"
          >
            <h1
              class="is-marginless"
            >
              Repo
            </h1>
          </div>
        </div>
      </div>
       
      <div
        class="container view-body"
      >
        <section>
          <div
            class="columns is-gapless"
          >
            <aside
              class="column is-one-quarter"
            >
              <div
                class="level"
              >
                <div
                  class="level-left"
                >
                  <div
                    class="field has-addons"
                  >
                    <div
                      class="control"
                    >
                      <a
                        class="button is-small"
                        href="#"
                      >
                        Lint
                      </a>
                    </div>
                     
                    <div
                      class="control"
                    >
                      <a
                        class="button is-small"
                        href="#"
                      >
                        Sync
                      </a>
                    </div>
                  </div>
                </div>
                 
                <div
                  class="level-right"
                >
                  <!---->
                   
                  <span
                    class="tag is-warning"
                  >
                    Unvalidated
                  </span>
                   
                  <!---->
                </div>
              </div>
               
              <!---->
               
              <!---->
               
              <p
                class="menu-label"
              >
                
              Topics
            
              </p>
               
              <ul
                class="menu-list"
              >
                <li>
                  <div
                    class="columns is-vcentered"
                  >
                    <div
                      class="column"
                    >
                      <a
                        class="is-active js-topics-carbon"
                      >
                        
                        carbon
                      
                      </a>
                    </div>
                     
                    <!---->
                  </div>
                </li>
              </ul>
            </aside>
             
            <div
              class="column is-three-quarters"
            >
              <!---->
               
              <div
                class="js-code-preview is-paddingless code-container"
              >
                <div
                  class="has-background-white"
                >
                  <pre>
                    {
  "title": "Title"
}
                  </pre>
                </div>
              </div>
            </div>
          </div>
        </section>
      </div>
    </div>
  </section>
</div>
`;
